src/Shared.TextureContainer/Shared.TextureContainer.cs

```diff
@@ -9,7 +9,7 @@ namespace KK_Plugins
     /// </summary>
     public sealed class TextureContainer : IDisposable
     {
-        private TextureContainerManager.Token _token;
+        internal TextureContainerManager.Token _token;
```

Comment:
This is naughty. If you need the hash then expose just the hash in a new method or prop, eg GetTextureHash.